lib/rules/require-await.js

```diff
@@ -42,14 +42,47 @@ module.exports = {
         schema: [],
 
         messages: {
-            missingAwait: "{{name}} has no 'await' expression."
-        }
+            missingAwait: "{{name}} has no 'await' expression.",
+            removeAsync: "Remove 'async'."
+        },
+
+        hasSuggestions: true
     },
 
     create(context) {
         const sourceCode = context.sourceCode;
         let scopeInfo = null;
 
+        /**
+         * Finds the range that covers the `async` keyword.
+         * @param {FunctionNode} node The function node to find the `async` keyword in.
+         * @returns {[number, number] | null} The range of the `async` keyword, or null if it could not be found.
+         */
+        function findAsyncKeywordRange(node) {
+
+            /*
+             * If the function belongs to a method definition or
+             * property, then the function's range will not include the
+             * `async` keyword. We need to look at the parent instead.
+             */
+            const nodeWithAsyncKeyword =
+                node.parent.type === "MethodDefinition" ||
+                node.parent.type === "Property"
+                    ? node.parent
+                    : node;
```

Comment:
```suggestion
            const nodeWithAsyncKeyword =
                (node.parent.type === "MethodDefinition" && node.parent.value === node) ||
                (node.parent.type === "Property" && node.parent.method && node.parent.value === node)
                    ? node.parent
                    : node;
```

In this case:

```js
/* eslint require-await: "error" */

const obj = {
    async: async function foo() {
        bar();
    }
}
```

suggestion would remove the wrong `await`:

```js
/* eslint require-await: "error" */

const obj = {
    : async function foo() {
        bar();
    }
}
```

It would be good to also add a test case with his example.